dotcom-rendering/scripts/dotenv/gen-dotenv.js

```diff
@@ -11,77 +11,99 @@ const checkEnv = async () => {
 	try {
 		const env = (await fs.readFile(ENV_PATH)).toString();
 
-		let valid = true;
 		for (const secret of secrets) {
-			const regex = new RegExp(`^${secret.key.replace('.', '\\.')}=.*`);
-			if (!env.match(regex)) valid = false;
+			const regex = new RegExp(`^${secret.key.replace('.', '\\.')}=.+`);
+			if (!env.match(regex)) return false;
 		}
 
-		return valid;
+		return true;
 	} catch (_err) {
 		return false;
 	}
 };
 
-const genEnv = async () => {
-	const env = process.env.NODE_ENV === 'production' ? 'prod' : 'dev';
-	const parameters = await getAwsSsmParameters(env);
-
+const genEnv = async (parameters) => {
 	let envString = '';
 	for (const secret of secrets) {
 		envString += `${secret.key}=${parameters[secret.key]}\n`;
 	}
 
 	await fs.writeFile(ENV_PATH, envString);
+
+	prompt('[scripts/dotenv] Generated .env file');
 };
 
-// eslint-disable-next-line @typescript-eslint/no-floating-promises
-(async () => {
+const prod = async () => {
 	try {
-		const validEnv = await checkEnv();
-		if (!validEnv) {
-			log(
-				'[scripts/dotenv] .env file is missing, attemting to generate it from AWS parameters...',
+		const params = await getAwsSsmParameters('prod');
```

Comment:
Is this always going to be 'prod'?

Reply:
Because we've separated out the functions which are called, e.g `prod` and `dev`, we can hardcode these values here. 

There may be a future where we want to also have 'code', however for this we'd need to introduce a new environment variable passed that isn't NODE_ENV (e.g GU_ENV) which isn't in the scope of this PR

Reply:
Gotcha. Thanks @OllysCoding 

Having `CODE` marked up as `prod` is an existing pattern that I'm happy for this PR to continue to support.

Outside of this PR, I do quite like how `CODE` can be considered production like but I also see how this can reduce flexibility and the ability to customise things.